openhtf/io/output/print_console.py

```diff
@@ -0,0 +1,81 @@
+"""Module for outputting test record to JSON-formatted files."""
+import os
+import sys
+from json import JSONEncoder
+from openhtf import util
+
+
+class PrintConsole(JSONEncoder):
+  """Return an output callback that writes JSON Test Records.
+
+  An example filename_pattern might be:
+    '/data/test_records/%(dut_id)s.%(start_time_millis)s'
+
+  To use this output mechanism:
+    test = openhtf.Test(PhaseOne, PhaseTwo)
+    test.AddOutputCallback(openhtf.OutputToJson(
+        '/data/test_records/%(dut_id)s.%(start_time_millis)s'))
+
+  Args:
+    filename_pattern: A format string specifying the filename to write to,
+      will be formatted with the Test Record as a dictionary.
+    inline_attachments: Whether attachments should be included inline in the
+      output.  Set to False if you expect to have large binary attachments.
+  """
+
+  def __init__(self, filename_pattern=None, inline_attachments=True, **kwargs):
+    super(PrintConsole, self).__init__(**kwargs)
+    self.FAIL =  '\033[91m'
+    self.PASS =  '\033[92m'
+    self.ERROR = '\033[93m'
+    self.ENDC =  '\033[0m'
+    self.BOLD =  '\033[1m'
+
+  # pylint: disable=invalid-name
+  def __call__(self, test_record):
+    as_dict = util.convert_to_dict(test_record)
```

Comment:
once #168 goes in, this will be an Enum that would be better to use directly.  In fact, you could build a map from enum value to color:
```
from openhtf.io import test_record

class PrintConsole(object):
  GREEN = '\033[92m'
  OUTCOME_COLOR_MAP = {
    test_record.Outcome.PASS: GREEN,
    ...
  }
  ...
  def __call__(self, test_record):
    ...
    print '\n' + self._colorize(
        self.OUTCOME_COLOR_MAP[test_record.outcome] + self.BOLD,
        os.path.basename(sys.argv[0]) + ': ' + test_record.outcome.name
    )
    ...